### ISSUE TITLE
[New Feature]: Package SPS w/ Airflow for the Unity MarketPlace

### ISSUE THREAD

**Issue by LucaCinquini** [unity-sds/unity-sps#49]
Acceptance Criteria:
o Demonstrated deployment of SPS w/ Airflow using the Unity Management Console

**Comment by jpl-btlunsfo** [comment]
Being blocked by differing versions of terraform in the unity-management console ([here](https://github.com/unity-sds/cfn-ps-jpl-unity-sds/blob/main/templates/unity-mc.main.template.yaml#L389)), and requirements in the unity-sps repo ([here](https://github.com/unity-sds/unity-sps/blob/main/terraform-unity/versions.tf) and [here](https://github.com/unity-sds/unity-sps/blob/main/terraform-unity/modules/terraform-eks-cluster/versions.tf)). 

Looking into updating unity-management-console's cloudformation templates to pull a newer terraform.

**Comment by jpl-btlunsfo** [comment]
After a number of little fixes to the sps eks deployment terraform (in branch 114-package-for-marketplace), I was finally able to get the management console actually running it.

The EKS cluster _seems_ to have installed successfully, but the other bits (helm charts, etc) seem to have failed/timed out after that. I'm not sure if that's a connectivity error (that the terraform is talking to the wrong endpoint), or just a timing issue- but I'm still looking into it.
```[ERROR] 
[ERROR] Error: Kubernetes cluster unreachable: Get "https://7AE228734033540E79AA2552C14882D4.sk1.us-west-2.eks.amazonaws.com/version": dial tcp 10.6.51.219:443: i/o timeout
[ERROR] 
[ERROR]   with module.spseks-qcdUP.module.unity-eks.helm_release.aws-load-balancer-controller,
[ERROR]   on .terraform/modules/spseks-qcdUP.unity-eks/terraform-unity-eks_module/main.tf line 423, in resource "helm_release" "aws-load-balancer-controller":
[ERROR]  423: resource "helm_release" "aws-load-balancer-controller" {
[ERROR] 
[ERROR] 
[ERROR] Error: Post "https://7AE228734033540E79AA2552C14882D4.sk1.us-west-2.eks.amazonaws.com/api/v1/namespaces/kube-system/serviceaccounts": context deadline exceeded
[ERROR] 
[ERROR]   with module.spseks-qcdUP.module.unity-eks.kubernetes_service_account.aws-load-balancer-controller-service-account,
[ERROR]   on .terraform/modules/spseks-qcdUP.unity-eks/terraform-unity-eks_module/main.tf line 445, in resource "kubernetes_service_account" "aws-load-balancer-controller-service-account":
[ERROR]  445: resource "kubernetes_service_account" "aws-load-balancer-controller-service-account" {
[ERROR]
```

Additionally, I wanted to note that neither the MC uninstall button on the application, nor for the entire MC instance seem to have cleaned up/done anything. Additionally, the MC teardown (`destroy.sh`) script likewise didn't seem to cleanup the created EKS cluster (or any of the other resources, I'd expect). Will something else clean those up, or is this something the management console _should_ be handling?

**Comment by GodwinShen** [comment]
@jpl-btlunsfo ping for status

**Comment by jpl-btlunsfo** [comment]
SPS (a slightly modified copy of the 2.2.0 release) deployed successfully in `unity-venue-dev`!

Should be available here:
- http://btl-dd6-dev-httpd-alb-1570456565.us-west-2.elb.amazonaws.com:8080/btl-dd6/dev/sps/
  - (user: admin, pw:\<slack me\>)
  - as deployed by the [`btl-dd6`/`dev` personal management console](http://btl-dd6-dev-httpd-alb-1570456565.us-west-2.elb.amazonaws.com:8080/btl-dd6/dev/management/ui/applications) instance
  
Required a bit of adjustment to get working, but the changes should be easy enough to bring forward into future releases.
Currently deployed from these branches:
- https://github.com/unity-sds/unity-sps/tree/59-marketplace-adjustments-2.2.0
- https://github.com/unity-sds/unity-cs-infra/tree/59-sps-eks-marketplace-adjustments
  - really just some generic additions to the `[terraform-unity-eks_module](https://github.com/unity-sds/unity-cs-infra/tree/59-sps-eks-marketplace-adjustments/terraform-unity-eks_module)` to allow communication from the management console
- https://github.com/unity-sds/unity-proxy/tree/59-temporarily-change-provider-version
   - just ensuring the proxy is pinned to the same aws module version that the SPS terraform uses (5.50.0) to prevent state issues on install.
   
"New" marketplace applications (in install order) for this:
 - [unity-sps-eks 0.1](https://github.com/unity-sds/unity-marketplace/tree/main/applications/unity-sps-eks/0.1/metadata.json)
 - [unity-sps-karpenter 0.1](https://github.com/unity-sds/unity-marketplace/tree/main/applications/unity-sps-karpenter/0.1/metadata.json)
 - [unity-sps 0.2](https://github.com/unity-sds/unity-marketplace/tree/main/applications/unity-sps/0.2/metadata.json)

I *have not yet tested any DAGs* in the airflow instance, but all of the installation steps run without issue.
(is this issue a duplicate of unity-sds/unity-sps#59?)
